nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java

```diff
@@ -102,9 +104,9 @@ private void resetData() throws NessieConflictException, NessieNotFoundException
   }
 
   @BeforeEach
-  public void beforeEach() throws IOException {
-    uri = server.getURI().toString();
-    this.api = HttpClientBuilder.builder().withUri(uri).build(NessieApiV1.class);
+  public void beforeEach(@NessieUri URI nessieUri) throws IOException {
```

Comment:
Question: What does this do / how does this work? Anything you can point me to (PR, etc) so I can peruse it myself?

Reply:
That was part of some refactoring for the test-related code, actually: make Jersey correctly expose the "right" base URI, as there was a bug when multiple tests using that Jersey-based code ran concurrently.
Previously there was a getter-method on the "server" class, but that became a parameter being filled from the JUnit5 extension (hence the `@NessieUri` annotation).

Reply:
just for completeness for @kbendick, this was introduced in https://github.com/projectnessie/nessie/pull/3073